### PR TITLE
Sailthru M2 SOW 4 Change request

### PR DESCRIPTION
**Pull request by giraclion**
- Send Magento 2 Admin Emails
- Sync Product Data Nightly to Sailthru
- Configurable Product Attributes for Product Sync

### CODE REVIEW

File: etc/adminhtml/system.xml [hunk 1]
```diff
@@ -63,30 +52,32 @@
...
-                    </comment>
+                    <comment><![CDATA[Should variant products be sent to Sailthru. NOTE: If variants aren't visible individually, must enable URL-fragmenting in Sailthru.]]></comment>
                     <depends>
                         <field id="*/*/enable_intercept">1</field>
                     </depends>
                 </field>
+                <field id="enable_intercept_cron" type="select" translate="label comment" sortOrder="3" showInDefault="1" showInWebsite="1" showInStore="1">
+                    <label>Enable Scheduled Products Sync to Sailthru</label>
+                    <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
+                    <comment><![CDATA[Sync Products also Nightly to Sailthru]]></comment>
```
Comment (silverman63) [inline etc/adminhtml/system.xml]: Wording

File: Helper/Api.php [hunk 2]
```diff
@@ -184,7 +189,7 @@ public function tagsUseKeywords()
         return $this->getSettingsVal(self::XML_CONTENT_USE_KEYWORDS);
     }
 
-    public function tagsUseAttributes()
+    public function isProductAttributesUsedForTagsVars()
     {
         return $this->getSettingsVal(self::XML_CONTENT_USE_ATTRIBUTES);
```
Comment (brownac) [inline Helper/Api.php]: prefer to have this as a `boolval` as well.

File: Helper/ProductData.php [hunk 3]
```diff
@@ -80,27 +81,42 @@ public function __construct(
...
+     *
      * @return bool
      */
     public function isProductInterceptOn($storeId = null)
     {
         return boolval($this->getSettingsVal(self::XML_CONTENT_INTERCEPT, $storeId));
     }
 
+    /**
+     * Is Synchtonization Product by Cron Enable
```
Comment (silverman63) [inline Helper/ProductData.php]: Spelling

File: etc/adminhtml/system.xml (hunk 1 above)
Comment (giraclion) [reply etc/adminhtml/system.xml]: @silverman63 let me know what to put here

File: Helper/Api.php (hunk 2 above)
Comment (giraclion) [reply Helper/Api.php]: @brownac this is a deprecated class, we are not utilizing it in main logic

File: Helper/ProductData.php (hunk 3 above)
Comment (giraclion) [reply Helper/ProductData.php]: @silverman63 fixed

**Pull request merged**